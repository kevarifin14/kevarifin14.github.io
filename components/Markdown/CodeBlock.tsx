import { ReactNode } from 'react';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';

type CodeBlockProps = {
  language: string,
  value: ReactNode,
}

export default function CodeBlock({ language, value }: CodeBlockProps) {
  return <SyntaxHighlighter language={language}>{value}</SyntaxHighlighter>;
}
