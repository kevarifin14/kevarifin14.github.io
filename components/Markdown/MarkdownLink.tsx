import { ReactNode } from 'react';

type MarkdownLinkProps = {
  href: string,
  children: ReactNode,
}

export default function MarkdownLink({ href, children }: MarkdownLinkProps) {
  return <a href={href} target="_blank" rel="noreferrer" className="text-primary">{children}</a>;
}
