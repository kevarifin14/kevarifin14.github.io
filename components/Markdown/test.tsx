import renderer from 'react-test-renderer';

import Markdown from '.';

describe('Markdown', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Markdown />,
    );
    expect(tree).toMatchSnapshot();
  });
});
