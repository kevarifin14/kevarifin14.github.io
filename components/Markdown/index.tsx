import ReactMarkdown from 'react-markdown';

import CodeBlock from './CodeBlock';
import MarkdownLink from './MarkdownLink';

type MarkdownProps = {
  source: string
};

export default function Markdown({ source }: MarkdownProps) {
  return (
    <ReactMarkdown
      escapeHtml={false}
      className="max-w-screen-md w-full prose mx-auto"
      source={source}
      renderers={{ code: CodeBlock, link: MarkdownLink }}
    />
  );
}
