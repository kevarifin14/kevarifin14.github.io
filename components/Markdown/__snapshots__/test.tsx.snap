// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown matches snapshot 1`] = `
<div
  className="max-w-prose w-full prose mx-auto"
/>
`;
